[Instrumentation.AspNetCore] .NET6 library works on .NET7 and .NET8 (#5252)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md

  [#4466](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4466)
  where the activity instance returned by `Activity.Current` was different than
  instance obtained from `IHttpActivityFeature.Activity`.
  ([#5136](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5136))

> * Fixed metrics instrumentation when library targeted for .NET6 or .NET 7

Reviewer: ```suggestion
* Fixed metrics instrumentation when library targets .NET 6 or .NET 7
```

>   was loaded by .NET8.

Reviewer: ```suggestion
  was loaded by .NET 8.
```

>   Fixed traces instrumentation when library targeted for .NET6
>   was loaded by .NET7 or .NET8.

Author: ```suggestion
  Fixed traces instrumentation when library targeting .NET 6
  was loaded by .NET 7 or .NET 8.
```
